cmd/translatesfx/translatesfx/sa.go

```diff
@@ -0,0 +1,226 @@
+package translatesfx
+
+import (
+	"fmt"
+	"os"
+	"path/filepath"
+	"strings"
+
+	"gopkg.in/yaml.v2"
+)
+
+// expandSA takes an unmarshalled Smart Agent config struct and returns a config
+// with any SA #from directives translated into their Otel equivalent.
+func expandSA(in interface{}, wd string) (interface{}, bool) {
+	switch t := in.(type) {
+	case []interface{}:
+		return expandSlice(t, wd)
+	case map[interface{}]interface{}:
+		return expandMap(t, wd)
+	default:
+		return in, false
+	}
+}
+
+func expandSlice(l []interface{}, wd string) (interface{}, bool) {
+	var out []interface{}
+	for _, v := range l {
+		next, flatten := expandSA(v, wd)
+		if flatten {
+			if a, ok := next.([]interface{}); ok {
+				for _, subV := range a {
+					out = append(out, subV)
+				}
+			}
+		} else {
+			out = append(out, next)
+		}
+	}
+	return out, false
+}
+
+func expandMap(m map[interface{}]interface{}, wd string) (interface{}, bool) {
+	if _, ok := m["#from"]; ok {
+		replacement, flatten, err := processDirective(m, wd)
+		if err == nil {
+			return replacement, flatten
+		}
+		fmt.Printf("unable to process directive: %v: error: %v\n", m, err)
+	}
+	out := map[interface{}]interface{}{}
+	for k, v := range m {
+		expanded, flatten := expandSA(v, wd)
+		if flatten {
+			if flattened, ok := expanded.(map[interface{}]interface{}); ok {
+				for fk, fv := range flattened {
+					out[fk] = fv
+				}
+			}
+		} else {
+			out[k] = expanded
+		}
+	}
+	return out, false
+}
+
+func processDirective(directive map[interface{}]interface{}, wd string) (interface{}, bool, error) {
+	expanded, err := expandFromSource(directive, wd)
+	if err != nil {
+		return nil, false, err
+	}
+
+	return expanded, flatten(directive), err
+}
+
+func expandFromSource(directive map[interface{}]interface{}, wd string) (interface{}, error) {
```

Comment:
There are a number of existing config sources and I wonder if it'd be better to replace the agent sources with ours rather than reimplement their logic like is done w/ file: https://github.com/signalfx/splunk-otel-collector/tree/main/internal/configsource

Reply:
^ This could require a new agent config source though...

Reply:
Yeah, I'm not sure what to do about that yet. One problem, I think, with using the configsource's file expansion capabilities is that SA supports globbing and merging, and configsource apparently doesn't. I suppose we might want to glob/merge then expand the result into a single file and create a configsource for that? Maybe suitable for a separate PR?